src/garage/torch/modules/gaussian_mlp_module.py

```diff
@@ -0,0 +1,223 @@
+            are two options:
+            - exp: the logarithm of the std will be stored, and applied a
+               exponential transformation
+            - softplus: the std will be computed as log(1+exp(x))
+        layer_normalization (bool): Bool for using layer normalization or not.
+    """
+
+    def __init__(self,
+                 input_dim,
+                 output_dim,
+                 hidden_sizes=(32, 32),
+                 hidden_nonlinearity=torch.tanh,
+                 hidden_w_init=nn.init.xavier_uniform_,
+                 hidden_b_init=nn.init.zeros_,
+                 output_nonlinearity=None,
+                 output_w_init=nn.init.xavier_uniform_,
+                 output_b_init=nn.init.zeros_,
+                 learn_std=True,
+                 adaptive_std=False,
+                 std_share_network=False,
+                 init_std=1.0,
+                 min_std=1e-6,
+                 max_std=None,
+                 std_hidden_sizes=(32, 32),
+                 std_hidden_nonlinearity=torch.tanh,
+                 std_hidden_w_init=nn.init.xavier_uniform_,
+                 std_hidden_b_init=nn.init.zeros_,
+                 std_output_nonlinearity=None,
+                 std_output_w_init=nn.init.xavier_uniform_,
+                 std_parameterization='exp',
+                 layer_normalization=False):
+        super(GaussianMLPModule, self).__init__()
+
+        self._input_dim = input_dim
+        self._hidden_sizes = hidden_sizes
+        self._action_dim = output_dim
+        self._learn_std = learn_std
+        self._adaptive_std = adaptive_std
+        self._std_share_network = std_share_network
+        self._std_hidden_sizes = std_hidden_sizes
+        self._min_std = min_std
+        self._max_std = max_std
+        self._std_hidden_nonlinearity = std_hidden_nonlinearity
+        self._std_hidden_w_init = std_hidden_w_init
+        self._std_hidden_b_init = std_hidden_b_init
+        self._std_output_nonlinearity = std_output_nonlinearity
+        self._std_output_w_init = std_output_w_init
+        self._std_parameterization = std_parameterization
+        self._hidden_nonlinearity = hidden_nonlinearity
+        self._hidden_w_init = hidden_w_init
+        self._hidden_b_init = hidden_b_init
+        self._output_nonlinearity = output_nonlinearity
+        self._output_w_init = output_w_init
+        self._output_b_init = output_b_init
+        self._layer_normalization = layer_normalization
+
+        # Tranform std arguments to parameterized space
+        self._init_std_param, self._min_std_param, self._max_std_param = None, None, None
+
+        init_std = torch.tensor(init_std, dtype=torch.float)
+        if min_std:
+            min_std = torch.tensor(min_std, dtype=torch.float)
+        if max_std:
+            max_std = torch.tensor(max_std, dtype=torch.float)
+
+        if self._std_parameterization == 'exp':
+            self._init_std_param = init_std.log()
```

Comment:
Since this is a `torch.tensor`, not a `torch.nn.Parameter`, it won't be trained, even if `learn_std` is True.

Reply:
fixed